dropshot/src/from_map.rs

```diff
@@ -46,22 +59,68 @@ impl MapValue for String {
     }
 }
 
+// To handle headers, which are case-insensitive, we use a trait to adapt
+// deserialization of structs to be case-insensitive.
+
+trait Casing {
+    fn rename(fields: &'static [&'static str]) -> BTreeMap<String, String>;
```

Comment:
This could use some docs?

(e.g., what are the output "Key/Values" in the BTreeMap here?)